### PR TITLE
Add `dev` and `dev_native` features

### PR DESCRIPTION
**Pull request by benfrankel**
Fixes https://github.com/TheBevyFlock/bevy-template/issues/18.

`bevy/dynamic_linking` works for both web and native builds, so it goes in `dev`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,4 +16,38 @@ opt-level = 1
...
+bevy = "0.14"
+# Required for web builds.
+wasm-bindgen = { version = "0.2", optional = true }
+
+[features]
+default = [
+    # Default to a native dev build.
+    "native_dev",
+]
+core = []
```
Comment (MiniaczQ) [inline Cargo.toml:28]: Add a comment for what sort of features go here

**Review by MiniaczQ — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (benfrankel) [reply Cargo.toml]: Don't need this feature in retrospect, removed.

**Review by MiniaczQ — APPROVED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -16,4 +16,35 @@ opt-level = 1
 opt-level = 3
 
 [dependencies]
-bevy = "0.12"
+bevy = "0.14"
+# Required for web builds.
+wasm-bindgen = { version = "0.2", optional = true }
```
Comment (janhohenheim) [inline Cargo.toml]:
```suggestion

[target.'cfg(target_family = "wasm")'.dependencies]
wasm-bindgen = "0.2"
```

File: Cargo.toml [hunk 3]
```diff
@@ -16,4 +16,35 @@ opt-level = 1
...
+
+[features]
```
Comment (janhohenheim) [inline Cargo.toml:24]: Trim these to only `dev` and `dev_native`.

**Review by janhohenheim — CHANGES_REQUESTED** [review]

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**